package-res/ccc/core/base/slidingWindow.js

```diff
@@ -0,0 +1,233 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/*global axis_optionsDef:true*/
+  
+
+//CDF603
+
+def('pvc.visual.SlidingWindow', pvc.visual.OptionsBase.extend({
+
+    init: function(chart) {
+        this.base( chart, 'slidingWindow', 0, {byNaked: false} );
+    },
+
+    type: {
+        methods: {
+            props: ['interval','dimName','score', 'select']
+        }
+    },
+
+    methods: /** @lends pvc.visual.slidingWindow# */{
+        _initFromOptions: function() {
+            var o = this.option;
+            this.set({
+                interval :  o('Interval'),
+                dimName  :  o('DimName'),
+                score    :  o('Score'),
+                select   :  o('Select')
+            });
+        },
+
+        set: function(keyArgs) {
+
+            keyArgs = this._readArgs(keyArgs);
+
+            if(!keyArgs) {
```

Comment:
The way the function resulted in, this branch is not needed. You could test only `if(keyArgs)`.